Tests/InAppPurchasesServiceTests.swift

```
@@ -41,24 +51,16 @@ final class InAppPurchasesServiceTests: XCTestCase {
 
     func testFetchProductSKUs_WhenRequestFails_PublishesReceivedError() throws {
 
-        let expectedError = URLError(.badServerResponse)
-
-        let service = InAppPurchasesService(
-            configuration: .fixture(),
-            networking: NetworkingStub(returning: .failure(expectedError))
-        )
+        let expectation = XCTestExpectation(description: "Publishes received error")
 
-        let expectation = XCTestExpectation(description: "Publishes received URLError")
+        stubRemoteResponse("/wpcom/v2/iap/products", filename: "iap-products-success.json", status: 503)
 
         service.fetchProductSKUs()
             .sink(
                 receiveCompletion: { completion in
-                    guard case .failure(let error) = completion else {
+                    guard case .failure = completion else {
                         return
```

Does it make sense to use `XCTFail("Expected to fail")` here instead of `return`?

---

Fixed 05f9df7
I decided to trigger XCTFail only in receiveCompletion 👍